src/Microsoft.Extensions.Options/IOptions.cs

```diff
@@ -7,11 +7,20 @@ namespace Microsoft.Extensions.Options
     /// Used to retreive configured TOptions instances.
     /// </summary>
     /// <typeparam name="TOptions">The type of options being requested.</typeparam>
-    public interface IOptions<out TOptions> where TOptions : class, new()
+    public interface IOptions<TOptions> where TOptions : class, new()
```

Comment:
So the behavior difference between `IOptions` and `IOptionsSnapshot` is that IOptions will never change, its a cached singleton that never gets updated.  IOptionsSnapshot is a scoped lifetime snapshot of the IOptionsMonitor.CurrentValue taken when the service is constructed.  So consumers of this (like Controllers) would naturally get updated options each request.